apps/organisations/mixins.py

```diff
@@ -0,0 +1,76 @@
+from django import forms
+from django.urls import reverse
+from django.utils.html import mark_safe
+from django.utils.translation import gettext_lazy as _
+
+from apps.organisations.models import OrganisationTermsOfUse
+
+
+class OrganisationTermsOfUseMixin(forms.ModelForm):
+    organisation_terms_of_use = forms.BooleanField(
+        required=False,
+        help_text=_(
+            'You can still manage all your preferences on User Agreements.'
+        )
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
+        super().__init__(*args, **kwargs)
+
+        self.fields['organisation_terms_of_use'].label = \
+            self._get_terms_of_use_label
+
+        if OrganisationTermsOfUse.objects.filter(
+            user=self.user,
+            organisation=self.module.project.organisation
+        ).exists() and OrganisationTermsOfUse.objects.get(
+            user=self.user,
+            organisation=self.module.project.organisation
+        ).has_agreed is True:
+            del self.fields['organisation_terms_of_use']
+
+    def clean(self):
+        cleaned_data = super().clean()
+        if 'organisation_terms_of_use' in cleaned_data:
+            organisation_terms_of_use = \
+                cleaned_data.get('organisation_terms_of_use')
+            if not organisation_terms_of_use:
+                self.add_error('organisation_terms_of_use',
+                               _("Please agree on the organisation's terms "
+                                 "of use to be allowed to create content."))
+        return cleaned_data
+
+    def save(self, commit=True):
+        instance = super().save(commit=commit)
+        if commit and 'organisation_terms_of_use' in self.cleaned_data:
+            try:
+                agreed = OrganisationTermsOfUse.objects.get(
+                    user=self.user,
+                    organisation=self.module.project.organisation
+                )
+                agreed.has_agreed = \
+                    self.cleaned_data['organisation_terms_of_use']
+                agreed.save()
+            except OrganisationTermsOfUse.DoesNotExist:
+                OrganisationTermsOfUse.objects.create(
+                    user=self.user,
+                    organisation=self.module.project.organisation,
+                    has_agreed=self.cleaned_data['organisation_terms_of_use']
+                )
```

Comment:
Maybe this could be handled with get_or_create()?

Reply:
Ah ok, didn't know that! I thought that this is exactly what get_or_create() does, so try: get, except: create. But if this is more robust, lets leave it like that.

Reply:
Should we then also remove get_or_create in other places, you think?